src/main/java/prefs/Preferences.java

```diff
@@ -15,83 +17,134 @@
-     * This will overwrite it with a default configuration if it already exists.
+     * @param sessionConfigFileName The name of the session config file
+     * @param ignoreExisting True if existing config file is to be ignored.
      */
-    public static Preferences create(String configFileName) {
-        return new Preferences(configFileName, true);
+    private Preferences(String sessionConfigFileName, String userConfigFileName, boolean ignoreExisting) {
+        this.sessionConfigFileName = sessionConfigFileName;
+        this.userConfigFileName = userConfigFileName;
+
+        this.sessionConfig = (SessionConfig) loadConfig(sessionConfigFileName, SessionConfig.class, ignoreExisting);
+        this.userConfig = (UserConfig) loadConfig(userConfigFileName, UserConfig.class, ignoreExisting);
     }
 
-    public void saveGlobalConfig() {
-        fileHandler.saveGlobalConfig(global);
+    /**
+     * Loads a Config instance from a file on disk
+     * @param configFileName The filename of the config file on disk
+     * @param configClass The class of the config to load
+     * @param ignoreExisting True if ignore the config file already
+     * @return The loaded Config instance
+     */
+    private Config loadConfig(String configFileName, Class<? extends Config> configClass, boolean ignoreExisting) {
+        FileHelper configFileHelper = new FileHelper(DIRECTORY, configFileName);
+        return new JsonHelper().createConfigFromJson(configFileHelper.loadFileContents(ignoreExisting), configClass);
     }
 
-    private void initGlobalConfig() {
-        global = fileHandler.initGlobalConfig();
+    /**
+     * Saves the session and user configs to file
+     */
+    public void flush() {
+        flushConfig(sessionConfig, sessionConfigFileName);
+        flushConfig(userConfig, userConfigFileName);
     }
 
-    private void loadGlobalConfig() {
-        global = fileHandler.loadGlobalConfig();
+    /**
+     * Saves a Config to file
+     * @param config The config object to save
+     * @param configFileName The file name of the config file on disk
+     */
+    private void flushConfig(Config config, String configFileName) {
+        JsonHelper jsonHelper = new JsonHelper();
+        String jsonString = jsonHelper.createJsonFromConfig(config);
+        FileHelper fileHelper = new FileHelper(DIRECTORY, configFileName);
+        fileHelper.writeFileContents(jsonString);
     }
 
     // Last login credentials. While the main UI is running (i.e. logged in successfully), last login
     // credentials are guaranteed to be the current user's credentials thanks to setLastLoginCredentials
     // being called immediately after a successful login in LoginDialog.
     public String getLastLoginPassword() {
-        return global.getLastLoginPassword();
+        return sessionConfig.getLastLoginPassword();
     }
 
+    /**
+     * Retrieves the last login username
+     */
```

Comment:
I'm not too sure where's the cut-off for self-explanatory, since Prof @damithc required header comments for methods. Do I remove the current comments for all the getters?